### ISSUE TITLE
Prometheus agent config validation error

### ISSUE THREAD

**Issue by kralicky** [rancher/opni#1540]
There appears to be an issue with our Prometheus CRD configuration that is causing Prometheus operator (v0.62) to deploy the prometheus agent (tried several versions: latest, 0.44, 0.43) with an invalid config. Specifically, on startup Prometheus fails to validate the config file with the error "field alerting is not allowed in agent mode".

**Comment by ron1** [comment]
Note that this issue appears to be negatively impacting the use of Opni within project Sylva. See https://gitlab.com/sylva-projects/sylva-core/-/merge_requests/646/diffs#2093015b6f93bd033624f214b05580dc015dd70d_0_454 for further details.

**Comment by kralicky** [comment]
@ron1 Thanks, I'll take a look. In the mean time, does the opentelemetry mode work instead?

**Comment by dbason** [comment]
We'll aim to get v0.11.1 out ASAP to fix this

**Comment by ron1** [comment]
@kralicky I suspect if opentelemetry mode was enabled by default, this would be a non-issue. It's always good for a project to give high-profile, early adopters the best out-of-the-box experience possible.

**Comment by alexandreLamarre** [comment]
@ron1 

FYI, there is a workaround that exists in the current opni version, using the CLI : 
```
opni metrics config set-default '{"rules":  {"discovery": { "prometheusRules": {} } },"otel":{"hostMetrics": false}}'
```

**Comment by dbason** [comment]
@ron1 I've been investigating this but we've had some issues reproducing it.  Our intention is to switch to OpenTelemetry as the default, however currently there are some problems with the RemoteWrite output of OpenTelemetry and we need to do some additional work to enable using OTLP through from agent to the gateway.

To assist us could you let us know what version of the prometheus operator your using when you see this?  Is it the bundled v0.62.0 or another version?  Also could you provide the Prometheus custom resource along with the generated prometheus config?

**Comment by kralicky** [comment]
Fixed by #1652
